pkg/registry/generic/registry/store.go

```diff
@@ -227,6 +230,22 @@ func (e *Store) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 	if err := e.Storage.Create(ctx, key, obj, out, ttl); err != nil {
 		err = storeerr.InterpretCreateError(err, e.QualifiedResource, name)
 		err = rest.CheckGeneratedNameError(e.CreateStrategy, err, obj)
+		if !storage.IsNodeExist(err) && e.OnCreateFailure == nil {
+			return nil, err
+		}
+		if errGet := e.Storage.Get(ctx, key, out, false); errGet != nil {
+			return nil, storeerr.InterpretGetError(errGet, e.QualifiedResource, name)
```

Comment:
You should return the original error here - this would break clients because they would suddenly get a new error.  From line 234 and onwards, you may only modify the original error's message - you cannot change the type of the error.